Doc/library/stdtypes.rst

```diff
@@ -622,6 +622,13 @@ class`. float also has the following additional methods.
    :exc:`OverflowError` on infinities and a :exc:`ValueError` on
    NaNs.
 
+   .. note::
+
+      The values returned by ``as_integer_ratio()`` can be huge. Attempts
```

Comment:
In case anyone gets worried by this, it may be worth noting that this _won't_ be a problem for Python `float`s on typical machines (i.e., those for which the format in use is IEEE 754 binary64, and I don't know of a single example where this isn't true for current CPython). For such machines, the largest possible integer that can result from `as_integer_ratio` has value `2**1074`, well within the default `4300` decimal digit limit.

Reply:
We can of course get huge numerators and denominators using `Decimal.as_integer_ratio` and `Fraction.as_integer_ratio`, but this doc edit is in the "Additional Methods on Float" section.

Noting also that `2**1074 < 10**640`, so that it wouldn't even be possible to trigger the limit by using `set_int_max_str_digits`, and that CPython now _requires_ IEEE 754 binary64 floating-point, I can't see any realistic way that we could bump into the `int_max_str_digits` limit with `as_integer_ratio`. I'd suggest reverting this addition.

Reply:
I left this in the docs as the minimum digit limit threshold someone could choose to set of 640 would allow code using python floats (assuming ieee754 double) to run into it.  In practice I found `as_integer_ratio` to be a very rarely used API in all code I could search over.

Reply:
edit: oh, you're right, even the lowest limit wouldn't hit it.  yay!